ramda.js

```diff
@@ -4705,6 +4707,42 @@
     R.gte = op(function _gte(a, b) { return a >= b; });
 
 
+    function createMaxMin(comparator, initialVal) {
+        return function(list) {
+            if (arguments.length === 0) {
+                throw noArgsException();
+            }
+            var idx = -1, winner = initialVal, computed;
+            while (++idx < list.length) {
+                computed = +list[idx];
+                if (comparator(computed, winner)) {
+                    winner = computed;
+                }
+            }
+            return winner;
+        };
+    }
+
+    function createMaxMinWith(comparator) {
+        return function(valueComputer, list) {
+            if (!(list && list.length > 0)) {
+                return;
```

Comment:
It does bother me a bit, too.  I simply don't see any good alternatives.  I think the signatures of `max` and of `maxBy` are correct.  But this means that `max` cannot be implemented as `maxBy(identity)` (which would be `[a] -> a` instead of `[Number] -> Number` (or our slightly relaxed version of that.)

I agree it does feel odd.  But anything that looks to fix it would likely cause something that seems to me much worse:

    max([a, b, ...]) !== max([b, a, ...); for some a, b

At that point this is no longer a number function and simply a partial ordering one.